integration/worker/dns_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/concourse/concourse/integration/internal/dctest"
+	"github.com/concourse/concourse/integration/internal/flytest"
+)
+
+func TestDNSProxyyEnabled(t *testing.T) {
+	t.Parallel()
+
+	dc := dctest.Init(t, "../docker-compose.yml", "overrides/enable_dns_proxy.yml")
+
+	t.Run("deploy with DNS proxy enabled", func(t *testing.T) {
+		dc.Run(t, "up", "-d")
+	})
+
+	fly := flytest.Init(t, dc)
+	fly.WithEnv("EXPECTED_EXIT_CODE=0").Run(t, "execute", "-c", "tasks/assert_web_is_reachable.yml")
+}
+
+func TestDNSProxyyDisabled(t *testing.T) {
+	t.Parallel()
+
+	dc := dctest.Init(t, "../docker-compose.yml", "overrides/disable_dns_proxy.yml")
+
+	t.Run("deploy with DNS proxy disabled", func(t *testing.T) {
+		dc.Run(t, "up", "-d")
+	})
+
+	fly := flytest.Init(t, dc)
+	fly.WithEnv("EXPECTED_EXIT_CODE=1").Run(t, "execute", "-c", "tasks/assert_web_is_reachable.yml")
+}
+
+func TestExtraDNSServersAreAdded(t *testing.T) {
+	t.Parallel()
+
+	dc := dctest.Init(t, "../docker-compose.yml", "overrides/add_extra_dns_servers.yml")
+
+	t.Run("deploy with extra DNS servers", func(t *testing.T) {
+		dc.Run(t, "up", "-d")
+	})
+
+	fly := flytest.Init(t, dc)
+	fly.WithEnv("EXTRA_SERVER=1.1.1.1").Run(t, "execute", "-c", "tasks/assert_extra_dns_servers.yml")
```

Comment:
Either way is good with me, but the assertion here could also be moved to the Go side of things:

```suggestion
	resolvConf := fly.Output(t, "execute", "-c", "tasks/cat_resolve_conf.yml")
	require.Contains(t, resolvConf, "1.1.1.1")
```

I realize I was the one who added the MTU tests following the pattern of these "assert" tasks with params to define the expected value 😁 but IMO it's a bit nicer pushing the assertions into Go in case the assertions become a bit more complicated. But again, no strong preference

Reply:
Making the tests dumber means we can probably share tasks more easily. We could just have a generic `cat` task that can be reused between integration tests, e.g.:

```go
resolvConf := fly.WithEnv("FILE=/etc/resolv.conf").Output(t, "execute", "-c", "tasks/cat.yml")
```

and in the MTU tests:

```go
mtu := fly.WithEnv("FILE=/sys/class/net/eth0/mtu").Output(t, "execute", "-c", "tasks/cat.yml")
```